plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/catalog/rest/TestIcebergBigLakeMetastoreConnectorSmokeTest.java

```diff
@@ -0,0 +1,377 @@
+import org.apache.iceberg.TableMetadataParser;
+import org.apache.iceberg.io.FileIO;
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.TestInstance;
+
+import java.io.IOException;
+import java.io.UncheckedIOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.Base64;
+import java.util.List;
+
+import static io.trino.plugin.iceberg.IcebergTestUtils.FILE_IO_FACTORY;
+import static io.trino.plugin.iceberg.IcebergTestUtils.checkParquetFileSorting;
+import static io.trino.testing.SystemEnvironmentUtils.requireEnv;
+import static io.trino.testing.TestingConnectorSession.SESSION;
+import static io.trino.testing.TestingNames.randomNameSuffix;
+import static java.lang.String.format;
+import static java.nio.charset.StandardCharsets.UTF_8;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+import static org.junit.jupiter.api.TestInstance.Lifecycle.PER_CLASS;
+
+@TestInstance(PER_CLASS)
+final class TestIcebergBigLakeMetastoreConnectorSmokeTest
+        extends BaseIcebergConnectorSmokeTest
+{
+    private static final String SCHEMA = "test_iceberg_biglake_" + randomNameSuffix();
+    private static final String GCP_STORAGE_BUCKET = requireEnv("GCP_STORAGE_BUCKET");
+    private static final byte[] GCS_JSON_KEY_BYTES = Base64.getDecoder().decode(requireEnv("GCP_CREDENTIALS_KEY"));
+    private static final ObjectMapper OBJECT_MAPPER = new ObjectMapperProvider().get();
+
+    public TestIcebergBigLakeMetastoreConnectorSmokeTest()
+    {
+        super(new IcebergConfig().getFileFormat().toIceberg());
+    }
+
+    @Override
+    protected boolean hasBehavior(TestingConnectorBehavior connectorBehavior)
+    {
+        return switch (connectorBehavior) {
+            case SUPPORTS_CREATE_MATERIALIZED_VIEW,
+                 SUPPORTS_RENAME_MATERIALIZED_VIEW,
+                 SUPPORTS_RENAME_SCHEMA,
+                 SUPPORTS_RENAME_TABLE -> false;
+            default -> super.hasBehavior(connectorBehavior);
+        };
+    }
+
+    @Override
+    protected QueryRunner createQueryRunner()
+            throws Exception
+    {
+        Path gcpCredentialsFile = Files.createTempFile("gcp-credentials", ".json");
+        gcpCredentialsFile.toFile().deleteOnExit();
+        Files.write(gcpCredentialsFile, GCS_JSON_KEY_BYTES);
+        String projectId = OBJECT_MAPPER.readTree(GCS_JSON_KEY_BYTES).get("project_id").asText();
+
+        return IcebergQueryRunner.builder(SCHEMA)
+                .addIcebergProperty("iceberg.file-format", format.name())
+                .addIcebergProperty("iceberg.register-table-procedure.enabled", "true")
+                .addIcebergProperty("iceberg.catalog.type", "rest")
+                .addIcebergProperty("iceberg.rest-catalog.uri", "https://biglake.googleapis.com/iceberg/v1beta/restcatalog")
```

Comment:
Can we document in https://trino.io/docs/current/object-storage/metastores.html#iceberg-rest-catalog how to connect to BigLake?
It may not be quite straightforward to figure out the minimal configuration setup needed if someone does not dive in the source code of `trinodb/trino`

FWIW, the same comment applies for other Iceberg REST Catalog integrations (e.g. : S3Tables, Unity, ...).

Reply:
Added an example to metastores.md. We already have Unity example. There is no other examples, though. 
https://github.com/trinodb/trino/blob/36b1485f3ffb6bc392df8d256a05c42d866e70b0/docs/src/main/sphinx/object-storage/metastores.md?plain=1#L541-L551